### ISSUE TITLE
Some leaks observed for twitter

### ISSUE THREAD

**Issue by alok0** [nikcub/Force-SSL#2]
I doubt this is fixable (but I don't know much about chrome's extension architecture), during special cases like login and initial entry into twitter.com, sometimes normal http requests are observed through my proxy.

**Comment by nikcub** [comment]
Yes it is fixable with the chrome.cookie API. I am going to make all site cookies secure cookies so that they are never transmitted over HTTP. It won't prevent the ordinary HTTP leaks, but at least those requests will be anonymous with no session or cookie data attached to them.

**Comment by alok0** [comment]
Also for facebook, I'd like to mention this [greasemonkey script](http://userscripts.org/scripts/review/86392) which has solved most of the links. Looks like it has something to do with hijacking something in facebook's code?
But it is able to convert links even if they are loaded via ajax.

**Comment by nikcub** [comment]
That is interesting. The method there is to find all the links in the page and re-write them. The way that I am building it now is to intercept any beforeload event in the page and if the URL that is about to be loaded is for any of the sites in options, it will rewrite it with SSL. I have tested this and it catches Ajax/XHR requests as well. I will push this code tonight - thanks for the feedback.
